src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+{
+    for (int i = 0; i < args->argc; i++) {
+        if (args->argv[i] != NULL) {
+            SCFree(args->argv[i]);
+            args->argv[i] = NULL;
+        }
+    }
+
+    SCFree(args->argv);
+    args->argv = NULL;
+    args->argc = 0;
+    args->capacity = 0;
+}
+
+int ArgumentsAdd(struct Arguments *args, char *value)
+{
+    if (args->argc + 1 > args->capacity) {
+        SCLogError(SC_ERR_DPDK_EAL_INIT, "No capacity for more arguments");
+        SCReturnInt(-ENOBUFS);
+    }
+
+    args->argv[args->argc++] = value;
+    SCReturnInt(0);
+}
+
+int ArgumentsAddOptionAndArgument(struct Arguments *args, const char *opt, const char *arg)
+{
+    int retval;
+    char *option;
+    char *argument;
+
+    option = AllocAndSetOption(opt);
+    if (option == NULL)
+        SCReturnInt(-errno);
+
+    retval = ArgumentsAdd(args, option);
+    if (retval < 0) {
+        SCFree(option);
+        SCReturnInt(retval);
+    }
+
+    // Empty argument could mean option only (e.g. --no-huge)
+    if (arg == NULL || arg[0] == '\0')
+        SCReturnInt(0);
+
+    argument = AllocAndSetArgument(arg);
+    if (argument == NULL) {
+        SCReturnInt(-errno);
+    }
+    ArgumentsAdd(args, argument);
+    SCReturnInt(0);
+}
+
+int InitEal()
+{
+    int retval;
+    ConfNode *param;
+    const ConfNode *ealParams = ConfGetNode("dpdk.eal-params");
+    struct Arguments args;
+    char **eal_argv;
+
+    if (ealParams == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF, "DPDK EAL parameters not found in the config");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    retval = ArgumentsInit(&args, EAL_ARGS);
+    if (retval < 0)
+        SCReturnInt(retval);
+
+    retval = ArgumentsAdd(&args, AllocAndSetArgument("suricata"));
+    if (retval < 0) {
+        SCReturnInt(retval);
+    }
+
+    TAILQ_FOREACH (param, &ealParams->head, next) {
+        retval = ArgumentsAddOptionAndArgument(&args, param->name, param->val);
+        if (retval < 0) {
+            ArgumentsCleanup(&args);
+            return retval;
+        }
+    }
+
+    // creating a shallow copy for cleanup because rte_eal_init changes array contents
+    eal_argv = SCMalloc(args.argc * sizeof(args.argv));
+    if (eal_argv == NULL) {
+        SCLogError(
+                SC_ERR_MEM_ALLOC, "Could not allocate memory for the array of DPDK EAL arguments");
+        errno = ENOMEM;
+        ArgumentsCleanup(&args);
+        SCReturnInt(-errno);
+    }
+    memcpy(eal_argv, args.argv, args.argc * sizeof(*args.argv));
+
+    rte_log_set_global_level(RTE_LOG_WARNING);
+    retval = rte_eal_init(args.argc, eal_argv);
+
+    ArgumentsCleanup(&args);
+    SCFree(eal_argv);
+
+    if (retval < 0) { // retval binded to the result of rte_eal_init
+        SCLogError(SC_ERR_DPDK_EAL_INIT, "DPDK EAL initialization error");
+        return retval;
+    }
+
+    if (rte_lcore_count() < 2) {
+        SCLogError(SC_ERR_DPDK_CONF, "Not enough cores configured. "
```

Comment:
user can provide DPDK cores with option `-c, -l, and --lcores`. Please reflect the real usage or point to DPDK documentation for the same. rework

Reply:
I do not udnerstand ` I try to reword it then.`, does this mean you will reword or are you waiting for something to reword?

Reply:
I initially have reworded it but in the end, I have completely deleted it from the next PR because this check became redundant. If not enough cores are configured for the given amount of queues, the Suricata stops later in the process of spawning lcores for the individual queues. If Suricata runs out of cores (because `lcore_nb < queue_nb`), then it shuts down.